### ISSUE TITLE
Store and provide license information

### ISSUE THREAD

**Issue by ethanwhite** [weecology/retriever#899]
We should have a license field for each script which stores the license information and should make it possible for users to obtain that information using commands like:

```
retriever license datasetname
```

and

```
retriever ls -l MIT GPL
```

**Comment by jainamritanshu** [comment]
@ethanwhite I want to work on this. Just a few clarifications. The basic idea to tackle this I can get is something like - 
- Adding a field `license` in the .json scripts
- adding a function `license` in datapackage.py 
```
def license(json_file):
    try:
        contents = json.load(
            open(os.path.join(HOME_DIR, 'scripts', json_file), 'r'))
    except (IOError,OSError):
        print("Script not found.")
        return

    if is_empty(contents['license'])
    	print ("\nNo license specified")
    	
    print ("\nMIT GPL")
```

Adding an argument clause of `license` in `__main__.py` and we will be good to go I guess. Kindly tell me if I am missing something.

**Comment by ethanwhite** [comment]
That's the basic idea.

The `datapackage.py` function's purpose is to get a license from the user creating a new JSON script, so it should read a license values using `clean_input` and assign it to the appropriate field to be written in the JSON.

The description of the license(s) should follow the datapackage spec. See the description under Properties -> Optional Properties -> Licenses at https://specs.frictionlessdata.io/data-package/#properties. It requires a link and we should also include the option license name and read that name for displaying when `retriever license` is called.

This might also require modifying the behavior of `retirever ls -l` so that it can match on licenses. I'm not sure if it will do this by default of will need to be modified.

**Comment by ethanwhite** [comment]
The core of this was implemented in #953. Closing and opening an issue focused on searching.
